internal/promotion/runner/builtin/git_cloner.go

```diff
@@ -77,7 +77,26 @@ func (g *gitCloner) Run(
 
 // validate validates gitCloner configuration against a JSON schema.
 func (g *gitCloner) validate(cfg promotion.Config) error {
-	return validate(g.schemaLoader, gojsonschema.NewGoLoader(cfg), g.Name())
+	err := validate(g.schemaLoader, gojsonschema.NewGoLoader(cfg), g.Name())
+	if err != nil {
+		return err
+	}
+
+	// Additional validation: ensure unique 'as' aliases in checkout
+	checkouts, ok := cfg["checkout"].([]promotion.Config)
+	if ok {
+		seen := make(map[string]struct{})
+		for i, c := range checkouts {
+			as, hasAs := c["as"].(string)
+			if hasAs && as != "" {
+				if _, exists := seen[as]; exists {
+					return fmt.Errorf("invalid git-clone config: duplicate checkout.as value %q at checkout[%d]", as, i)
+				}
+				seen[as] = struct{}{}
+			}
+		}
+	}
```

Comment:
All of this would be much cleaner if you did it on typed config.

You have a couple choices:

* Rename this method to `validateAndUnmarshal()` and have it make the `promotion.ConfigToStruct[builtin.GitCloneConfig](stepCtx.Config)` call between the two validation steps. The typed config can be returned along with the error. Then `promotion.ConfigToStruct...` does not need to be called separately in `Run()`.

* Make a separate method for these additional validation steps and call it from `Run()` after it's called `promotion.ConfigToStruct...`.

Reply:
I will try to do the first option- rename the method to `validateAndUnmarshal()` and perform both schema validation and struct unmarshalling inside it, i think thats the cleanest and most idiomatic as it will keep all config validation and conversion logic in one place and avoid double-parsing or awkward error handling in Run()